test/gapic-mocks/client-initialization-testing.ts

```diff
@@ -0,0 +1,165 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {beforeEach, describe, it} from 'mocha';
+import {
+  Datastore,
+  DatastoreClient,
+  Fallback,
+  DatastoreRequest,
+  DatastoreOptions,
+} from '../../src';
+import * as assert from 'assert';
+import * as proxyquire from 'proxyquire';
+import {Callback, CallOptions} from 'google-gax';
+import * as protos from '../../protos/protos';
+import * as ds from '../../src';
+import * as mocha from 'mocha';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Any = any;
+
+const clientName = 'DatastoreClient';
+const async = require('async');
+
+/**
+ * This class mocks out the lookup function so that for tests in this file
+ * the lookup function just sends data back instead of making a call to the
+ * server. The class also saves the rest parameter in the constructor so that
+ * it can be read later for correctness.
+ *
+ */
+class FakeDatastoreClient extends DatastoreClient {
+  restParameter: string | undefined;
```

Comment:
Hi Dan, just wondering is this "restParameter" expecting a boolean type from `Fallback type` too? https://github.com/googleapis/nodejs-datastore/pull/1203/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R100

Reply:
I changed the type to `Fallback` here so that the types line up better.